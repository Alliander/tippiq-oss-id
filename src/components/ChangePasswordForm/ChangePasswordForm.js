/**
 * Change password form component.
 * @module components/ChangePasswordForm
 */

import memoize from 'lru-memoize';
import React, { Component, PropTypes } from 'react';
import { reduxForm } from 'redux-form';
import { Row, Col } from 'react-bootstrap';

import Field from '../Field/Field';
import { createValidator, isRequired, isValidPassword } from '../../utils/validation';

 /**
 * ChangePasswordForm component.
 * @class ChangePasswordForm
 */
export class ChangePasswordForm extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    fields: PropTypes.object.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    errorMessage: PropTypes.string,
  }

/**
 * render
 * @function render
 * @returns {string} Markup of the ChangePassword component.
 */
  render() {
    const error = this.props.errorMessage ?
      <Col className="alert alert-danger">{this.props.errorMessage}</Col> : null;
    return (
      <Row>
        { error }
        <form onSubmit={this.props.handleSubmit}>
          <Col xs={12} sm={6} lg={5}>
            <Field
              field={this.props.fields.password}
              placeholder="Jouw nieuwe wachtwoord"
              type="password"
              id="password"
            />
          </Col>
          <Col xs={12} sm={6} lg={5}>
            <button
              id="passwordSubmitButton"
              type="submit"
              className="btn btn-primary btn-block"
              onClick={this.props.handleSubmit}
            >Opslaan</button>
          </Col>
        </form>
      </Row>
    );
  }
}


export default reduxForm({
  form: 'change-password',
  fields: ['password'],
  validate: memoize(10)(createValidator({
    password: [isRequired, isValidPassword],
  })),
})(ChangePasswordForm);
