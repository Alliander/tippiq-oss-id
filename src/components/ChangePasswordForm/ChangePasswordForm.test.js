import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ChangePasswordForm } from './ChangePasswordForm';

test('Basic render snapshot test', () => {
  const component = shallow(<ChangePasswordForm
    fields={{ password: 'password' }}
    handleSubmit={() => {}}
    errorMessage={'errormessage'}
  />);

  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

