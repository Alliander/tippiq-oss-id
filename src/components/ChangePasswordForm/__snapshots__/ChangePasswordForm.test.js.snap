// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic render snapshot test 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    className="alert alert-danger"
    componentClass="div"
  >
    errormessage
  </Col>
  <form
    onSubmit={[Function]}
  >
    <Col
      bsClass="col"
      componentClass="div"
      lg={5}
      sm={6}
      xs={12}
    >
      <Field
        field="password"
        id="password"
        placeholder="Jouw nieuwe wachtwoord"
        type="password"
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      lg={5}
      sm={6}
      xs={12}
    >
      <button
        className="btn btn-primary btn-block"
        id="passwordSubmitButton"
        onClick={[Function]}
        type="submit"
      >
        Opslaan
      </button>
    </Col>
  </form>
</Row>
`;
